### ISSUE TITLE
KeyError in `IBMQJob::error_message()` while running a circuit that throws

### ISSUE THREAD

**Issue by atilag** [Qiskit/qiskit-ibmq-provider#113]
### Informations

- **Qiskit version**: 0.8
- **Python version**: All
- **Operating system**: All

### What is the current behavior?
This circuit:
```python
from qiskit import QuantumRegister, ClassicalRegister
from qiskit import QuantumCircuit, Aer, execute
from qiskit.providers.ibmq import IBMQ

qe_token = "<HIDDEN>"
qe_url =  "The.API.url"

q = QuantumRegister(50)
c = ClassicalRegister(2)
qc = QuantumCircuit(q, c)

qc.h(q[0])
qc.cx(q[0], q[1])
for k in range(2,50):
    qc.t(q[k])

IBMQ.enable_account(qe_token, qe_url)
backend = IBMQ.get_backend('ibmq_qasm_simulator')
job_sim = execute(qc, backend)
try:
  job_sim.result()
except Exception:
  pass

print(job_sim.error_message())
```
Won't execute because the number of Qubits is too high, but the IBMQ Provider seems to not handle the error properly, because if I try to get the error message like:
```
job_sim.error_message()
```
Python throws an exception complaining about the inexistence of this key: `qObjectResult`.

### Steps to reproduce the problem
Execute the provided example program

### What is the expected behavior?
 `job_sim.error_message()` should show the error message.

**Comment by diego-plan9** [comment]
Thanks @atilag (nice bug report)! It seems the result of #50, which hopefully is on its way of being fixed.

**Comment by diego-plan9** [comment]
Closed via #126
